atomicfu-gradle-plugin/src/main/kotlin/kotlinx/atomicfu/plugin/gradle/AtomicFUGradlePlugin.kt

```diff
@@ -129,6 +129,7 @@ private fun Project.configureMultiplatformPluginDependencies(version: String) {
     }
     // Include atomicfu as a dependency for publication when transformation for the target is disabled
     multiplatformExtension.targets.all { target ->
```

Comment:
A related nitpick: typically `all {}` should be avoided, or where possible, limited in its impact by using `matching {}`. 

`configureEach {}` is preferred, because it doesn't trigger the realisation of all the contents of a Gradle collection. In contrast `all {}` will eagerly configure all elements. Sometimes `all {}` is required - and since Atomicfu is adding a dependency I think this is one such case. Without it, the dependency might not be added in time. (I don't have a hard or fast rule for this - Gradle is mysterious sometimes!)

Because the `all {}` block contains some conditional logic, it might well eagerly configure a target unnecessarily. The impact can be avoided by moving the `if` logic into a `matching {}` filter.

For this function because there are two different if statements, that means two different `all {}` blocks. Something like this:

```kotlin
// Include atomicfu as a dependency for publication when transformation for the target is disabled
multiplatformExtension.targets
  .matching { target -> isTransitiveAtomicfuDependencyRequired(target) }
  .all { target ->
    // Add an implementation dependency for native/wasm targets or if transformation is disable
        target.compilations.all { compilation ->
            compilation
                .defaultSourceSet
                .dependencies {
                    implementation(atomicfuDependency)
                }
        }
    }
  }

// atomicfu should also appear in apiElements config for native targets, 
// otherwise the warning is triggered, see: KT-64109
multiplatformExtension.targets
  .matching { target -> target.platformType == KotlinPlatformType.native }
  .all { target ->
    target.compilations.all { compilation ->
        compilation
            .defaultSourceSet
            .dependencies {
                api(atomicfuDependency)
            }
    }
}
```


However, this will be a behaviour change that you might not want to implement in this PR. I'll leave it up to you!

Reply:
Thank you! That was a very useful comment. I think, I'll better schedule this change to the separate PR just for the clear history.